### DISCUSSION TITLE
OSError: [Errno 9] Bad file descriptor happens

### DISCUSSION THREAD

**Discussion by pengyuzhang97** [microsoft/nni#4626] [Q&A] [answered]
Hello,

I was running the official mnist-pytorch code under Ubuntu 18.04.4 clone from examples/trials)/mnist-pytorch/ when this OSError happens. Don't really know how to fix it.

![}KAFWQNV)P%9M)S2J_2R6VO](https://user-images.githubusercontent.com/54292184/157381068-94cb0a3e-99d7-4e13-a642-2ac6a2d5b9b8.png)

**Comment by ultmaster** [comment, marked as answer]
You mentioned you ran the examples from `trials/mnist-pytorch`, but your log shows that you are running it from Retiarii. Which example are you using.

As far as I know, Retiarii is using an atexit, which might be not compatible with runners like pycharm at exit. Maybe you should try directly running it in a terminal.

**Reply by pengyuzhang97** [reply]
Sorry for the confusion. I was actually running the code under 'nni/examples/nas/mnist' which indeed uses the Retiarii. Now everything seems to work. Thank you for your answer.
